index.bs

```diff
@@ -4437,6 +4439,252 @@ The [=remote end subscribe steps=], with [=subscribe priority=] 10, given
+      InputPerformActionsCommand = {
+        method: "input.performActions",
+        params: InputPerformActionsParameters
+      };
+
+      InputPerformActionsParameters = {
+        context: BrowsingContext,
+        actions: [* InputSourceActions]
+      };
+
+      InputSourceActions = (
+        NoneSourceActions //
+        KeySourceActions //
+        PointerSourceActions //
+        WheelSourceActions
+      );
+
+      NoneSourceActions = {
+        type: "none",
+        actions: [* NoneSourceAction]
+      };
+
+      NoneSourceAction = PauseAction;
+
+      KeySourceActions = {
+        type: "key",
+        actions: [* KeySourceAction]
+      };
+
+      KeySourceAction = (
+        PauseAction //
+        KeyDownAction //
+        KeyUpAction
+      );
+
+      PointerSourceActions = {
+        type: "pointer",
+        ? parameters: PointerParameters,
+        actions: [* PointerSourceAction]
+      };
+
+      PointerParameters = {
+        ? pointerType: "mouse" / "pen" / "touch"
+      };
+
+      PointerSourceAction = (
+        PauseAction //
+        PointerDownAction //
+        PointerUpAction //
+        PointerMoveAction
+      );
+
+      WheelSourceActions = {
+        type: "wheel",
+        actions: [* WheelSourceAction]
+      };
+
+      WheelSourceAction = (
+        PauseAction //
+        WheelScrollAction
+      );
+
+      PauseAction = {
+        type: "pause",
+        ? duration: uint
+      };
+
+      KeyDownAction = {
+        type: "keyDown",
+        value: text
+      };
+
+      KeyUpAction = {
+        type: "keyUp",
+        value: text
+      };
+
+      PointerUpAction = {
+        type: "pointerUp",
+        button: uint,
+        PointerCommonProperties,
+      };
+
+      PointerDownAction = {
+        type: "pointerUp",
+        button: uint,
+        PointerCommonProperties
+      };
+
+      PointerMoveAction = {
+        type: "pointerMove",
+        x: uint,
+        y: uint,
+        ? duration: uint,
+        ? origin: Origin,
+        PointerCommonProperties,
+      };
+
+      WheelScrollAction = {
+        type: "scroll",
+        x: uint,
+        y: unint,
+        deltaX: int,
+        deltaY: int,
+        ? duration: uint,
+        ? origin: Origin,
+        PointerCommonProperties,
+      };
+
+      PointerCommonProperties = (
+        ? width: uint,
+        ? height: uint,
+        ? pressure: float,
+        ? tangentialPressure: float,
+        ? tiltX: int,
+        ? tiltY: int,
+        ? twist: uint,
+        ? altitudeAngle: float,
+        ? azimuthAngle: float,
+      )
+
+      Origin = "viewport" / "pointer" / ElementOrigin;
+
+      ElementOrigin = {
+        "type": "Element",
+        sharedId: SharedId
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+A map matching the <code>ElementOrigin</code> production [=represents a web
```

Comment:
The "represents a web element" term ends up not linked to anything. I guess it should be https://w3c.github.io/webdriver/#dfn-represents-a-web-element, but that doesn't quite make sense to me.

Should this say that an `Element`-relative origin is represented by `ElementOrigin`? (I don't think we need to say "matching a ... production" except where creating these structures before sending them in a response.)

Reply:
The broken link notwithstanding, this is not needed if we merge https://github.com/w3c/webdriver/pull/1653. The "problem" is that we don't want to copy all the actions text from WebDriver, we want to reuse it. But WebDriver defines an object as "representing a web element" when it has the magic property  "<code>element-6066-11e4-a52e-4f735466cecf</code>". That's an ugly hack that we don't need in BiDi where we already have a uniform JS value representation. So the problem we need to solve is to allow the BiDi representation of an element without replicating the entire actions section just to change one definition. The approach here was to say that for a BiDi implementation anything matching `ElementOrigin` is also something that "represents a web element" (assuming you implement BiDi), but the approach in the WebDriver PR is just to make the BiDi representation always work for the `origin` property in actions, even if called from HTTP. The latter is somewhat simpler at the cost of (theoretically at least) adding new requirements on existing implementations.